code/modules/research/mechanic/component_exchanger.dm

```diff
@@ -62,14 +62,18 @@
 		user.visible_message("<span class='notice'>[user] starts setting up \the [src] in \the [M]'s maintenance hatch</span>", \
 		"<span class='notice'>You carefully insert \the [src] through \the [M]'s maintenance hatch, it starts scanning the machine's components.</span>")
 
-		if(do_after(user, A, 20)) //2 seconds to obtain a complete reading of the machine's components
+		working = 1
+
+		if(do_after(user, A, 20)) //Two seconds to obtain a complete reading of the machine's components
 
 			if(!Adjacent(user))
```

Comment:
If user wasn't adjacent to the component exchanger do_after wouldnt return true
`if(!A.adjacent(user))`